modules/ding_frontend/ding_frontend.admin.inc

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/**
+ * @file
+ * Module's admin features.
+ */
+
+/**
+ * Administration form.
+ */
+function ding_frontend_settings($form, &$form_state) {
+  $types = array(
+    'ding_news' => t('News'),
+    'ding_event' => t('Event'),
+  );
+  $form['ding_frontend_fe_promote_types'] = array(
+    '#title' => t('Promote types to Fronpage'),
+    '#type' => 'checkboxes',
+    '#options' => $types,
+    '#default_value' => variable_get('ding_frontend_fe_promote_types', array('ding_event' => 'ding_event')),
+  );
+
+  $form = system_settings_form($form);
```

Comment:
I do not think using `system_settings_form()` makes much sense here. The variable could get out of sync with the linked variables and changing the strongarmed variable will not change the actual configuration.

This current submit function looks good so I suggest:

- Using the state of the `node_options` variables to determine default values 
- Not using `system_settings_form()``
- Removing the now obsolete `ding_frontend_fe_promote_types` variable from strongarm.